common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -70,120 +70,96 @@ public static String modifyMultipleMatchKeyword(String query){
 
         Specification<Concept> conceptSpecification =
                 (root, criteriaQuery, criteriaBuilder) -> {
                     List<Predicate> predicates = new ArrayList<>();
-
-                    // Check that the concept name, code, or ID matches the query string.
-                    List<Predicate> conceptCodeIDName = new ArrayList<>();
-                    Expression<Double> matchExp = null;
-
-                    if(!Strings.isNullOrEmpty(query)){
-
-                        final String keyword = modifyMultipleMatchKeyword(query);
-
-                        conceptCodeIDName.add(criteriaBuilder.equal(root.get("conceptCode"),
-                                criteriaBuilder.literal(query)));
-
-                        try {
-                            long conceptId = Long.parseLong(query);
-                            conceptCodeIDName.add(criteriaBuilder.equal(root.get("conceptId"),
-                                    criteriaBuilder.literal(conceptId)));
-                        } catch (NumberFormatException e) {
-                            // Not a long, don't try to match it to a concept ID.
+                    List<Predicate> standardConceptPredicates = new ArrayList<>();
+                    standardConceptPredicates.add(criteriaBuilder.equal(root.get("standardConcept"),
+                            criteriaBuilder.literal(STANDARD_CONCEPT_CODE)));
+                    standardConceptPredicates.add(criteriaBuilder.equal(root.get("standardConcept"),
+                            criteriaBuilder.literal(CLASSIFICATION_CONCEPT_CODE)));
+
+                    List<Predicate> nonStandardConceptPredicates = new ArrayList<>();
+                    nonStandardConceptPredicates.add(criteriaBuilder.notEqual(root.get("standardConcept"),
+                            criteriaBuilder.literal(STANDARD_CONCEPT_CODE)));
+                    nonStandardConceptPredicates.add(criteriaBuilder.notEqual(root.get("standardConcept"),
+                            criteriaBuilder.literal(CLASSIFICATION_CONCEPT_CODE)));
+
+                    // If the concepts that match the synonym names are found, they are fetched by applying standard concept filer, domain and vocabulary id filters
+
+                    if(synonymConceptIds.size() > 0){
+                        List<Predicate> synonymConceptPredicate = new ArrayList<>();
+                        Expression<Long> conceptIdCheck = root.get("conceptId");
+                        synonymConceptPredicate.add(conceptIdCheck.in(synonymConceptIds));
+                        if (standardConceptFilter.equals(StandardConceptFilter.STANDARD_CONCEPTS) || standardConceptFilter.equals(StandardConceptFilter.STANDARD_OR_CODE_ID_MATCH)) {
+                                predicates.add(criteriaBuilder.or(synonymConceptPredicate.toArray(new Predicate[0])));
```

Comment:
Just add the synonymPredicate once at the top before this if else stuff ? Looks like same line 3 times.